pkg/reduce/pbq/pbq.go

```diff
@@ -86,7 +88,13 @@ func (p *PBQ) CloseOfBook() {
 // Close is used by the writer to indicate close of context
 // we should flush pending messages to store
 func (p *PBQ) Close() error {
-	return p.store.Close()
+	p.mu.Lock()
```

Comment:
this is accessed by only one routine at any point in time right?

Reply:
i see. GC will cause the store to be set to nil, where as the store.Close() could be invoked. ok makes sense.